templates/default/source.conf.erb

```diff
@@ -2,7 +2,11 @@
 # Do NOT modify this file by hand!
 
 <source>
+  <% if node['td_agent']['version'].to_i < 3 %>
   type <%= @type %>
+  <% else %>
+  @type <%= @type %>
+  <% end %>
   <% if @tag %>
   tag <%= @tag %>
   <% end %>
```

Comment:
Please add `<= @parameters -%>` and remove the loop of `<% @parameters.each do |k,v| %>`.